storm-webapp/src/main/java/org/apache/storm/daemon/logviewer/LogviewerServer.java

```diff
@@ -0,0 +1,182 @@
+import org.apache.storm.ui.FilterConfiguration;
+import org.apache.storm.ui.UIHelpers;
+import org.apache.storm.utils.ConfigUtils;
+import org.apache.storm.utils.ObjectReader;
+import org.apache.storm.utils.Utils;
+import org.eclipse.jetty.server.Server;
+import org.eclipse.jetty.servlet.DefaultServlet;
+import org.eclipse.jetty.servlet.ServletContextHandler;
+import org.eclipse.jetty.servlet.ServletHolder;
+import org.eclipse.jetty.util.resource.Resource;
+import org.glassfish.jersey.servlet.ServletContainer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The main entry of Logviewer.
+ */
+public class LogviewerServer implements AutoCloseable {
+    private static final Logger LOG = LoggerFactory.getLogger(LogviewerServer.class);
+    private static final Meter meterShutdownCalls = StormMetricsRegistry.registerMeter("drpc:num-shutdown-calls");
+    public static final String STATIC_RESOURCE_DIRECTORY_PATH = "./public";
+
+    private static Server mkHttpServer(Map<String, Object> conf) {
+        Integer logviewerHttpPort = (Integer) conf.get(DaemonConfig.LOGVIEWER_PORT);
+        Server ret = null;
+        if (logviewerHttpPort != null && logviewerHttpPort >= 0) {
+            LOG.info("Starting Logviewer HTTP servers...");
+            Integer headerBufferSize = ObjectReader.getInt(conf.get(UI_HEADER_BUFFER_BYTES));
+            String filterClass = (String) (conf.get(DaemonConfig.UI_FILTER));
+            @SuppressWarnings("unchecked")
+            Map<String, String> filterParams = (Map<String, String>) (conf.get(DaemonConfig.UI_FILTER_PARAMS));
+            FilterConfiguration filterConfiguration = new FilterConfiguration(filterClass, filterParams);
+            final List<FilterConfiguration> filterConfigurations = Arrays.asList(filterConfiguration);
+
+            final Integer httpsPort = ObjectReader.getInt(conf.get(DaemonConfig.LOGVIEWER_HTTPS_PORT), 0);
+            final String httpsKsPath = (String) (conf.get(DaemonConfig.LOGVIEWER_HTTPS_KEYSTORE_PATH));
+            final String httpsKsPassword = (String) (conf.get(DaemonConfig.LOGVIEWER_HTTPS_KEYSTORE_PASSWORD));
+            final String httpsKsType = (String) (conf.get(DaemonConfig.LOGVIEWER_HTTPS_KEYSTORE_TYPE));
+            final String httpsKeyPassword = (String) (conf.get(DaemonConfig.LOGVIEWER_HTTPS_KEY_PASSWORD));
+            final String httpsTsPath = (String) (conf.get(DaemonConfig.LOGVIEWER_HTTPS_TRUSTSTORE_PATH));
+            final String httpsTsPassword = (String) (conf.get(DaemonConfig.LOGVIEWER_HTTPS_TRUSTSTORE_PASSWORD));
+            final String httpsTsType = (String) (conf.get(DaemonConfig.LOGVIEWER_HTTPS_TRUSTSTORE_TYPE));
+            final Boolean httpsWantClientAuth = (Boolean) (conf.get(DaemonConfig.LOGVIEWER_HTTPS_WANT_CLIENT_AUTH));
+            final Boolean httpsNeedClientAuth = (Boolean) (conf.get(DaemonConfig.LOGVIEWER_HTTPS_NEED_CLIENT_AUTH));
+
+            //TODO a better way to do this would be great.
```

Comment:
The comment was copied and pasted from DRPCServer.

While porting I thought about how to do it better, but no idea given that LogviewerApplication instance is generated dynamically. It may be possible to not passing config map and re-read storm configuration in LogviewerApplication, but it is duplicated effort and we will not be able to inject other config map in unit test. 

Maybe we could just remove the TODO comment.